draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -347,6 +347,41 @@ disjoint and non-colliding. Moreover, the length of each each public
 encapsulation key, ciphertext, and shared secret is fixed once the algorithm
 is assumed to be fixed.
 
+## Key generation and derivation {#generic-keygen}
+
+For both constructions we provide a common key generation and derivation
+design. It relies on the following parameters that are populated by concrete
+instantiations:
+
+* `XOF`: The eXtended Output Function
+* Nseed: length in bytes of the seed randomness sourced from the RNG
+* Nxofout: length in bytes of the XOF output
+* Npqseed: length in bytes of the input to PQ.DeriveKey()
+* Ntradseed: length in bytes of the input to NominalGroup.ScalarFromBytes()
+
+~~~
+def expandDecapsulationKey(sk):
+  expanded = XOF(sk, Nxofout)
```

Comment:
Maybe we replace `Nxofout` with `Npqseed + Ntradseed`? I think the `expanded[..Npqseed]` is fairly clear, but we should probably add some text to the terminology section that describes how it works. Something like:

```
When `x` is a byte string, we use the notation `x[..i]` and `x[i..]` to denote the slice of bytes in `x` starting from the beginning of `x` and leading up to index `i`, including the `i`-th byte, and the slice the bytes in `x` starting from index `i` to the end of `x`, respectively. For example, if `x = [0, 1, 2, 3]`, then `x[..2] = [0, 1]` and `x[2..] = [2, 3]`.
```

Reply:
Done; can you cross check the numbers for Ntradseed for the p256 and p384 instances?

Reply:
confirmed out of bad - they are larger than expected  to make sure the Scalar isn't biased (not a power of 2), these match HashToCurve